### PR TITLE
Fixed and changed segment traits

### PR DESCRIPTION
**Pull request by jordanh**
- Fixed: identify() events were occasionally passing the wrong
     HubSpot traits
   - Changed: renamed `name` trait to `parabolPreferredName`. `name`
     was apparently being interpreted by HubSpot as `firstName`. This
     was not desired. Now it's clear it's our own thing. No other
     services depending on Segment traits rely on the `name` field,
     so this change should be safe to make (they rely on `id` or
     `email`)

### CODE REVIEW

**Comment by jordanh** [comment]
## Tests

Starting app with segment dev keys

   - Navigate to http://localhost:3000/me/profile and change the user's name
   - [ ] can see identify() event segment debugger
   - [ ] `name` field not present, instead is named `parabolPreferredName` and contains updated value
   - [ ] can see trait `highestTier` and value looks correct

File: src/server/utils/sendSegmentEvent.ts [hunk 1]
```diff
@@ -112,14 +114,16 @@ export const sendSegmentIdentify = async (maybeUserIds) => {
...
   ])
-  traitsArr.forEach(async (traitsWithId, idx) => {
-    const {id: userId, ...traits} = traitsWithId
+  traitsArr.forEach(async (traitsWithId) => {
+    const {id: userId, ...traits} = {
+      ...traitsWithId,
+      ...hubspotTraitsArr.find((hubspotTraitsWithId) => traitsWithId.id === hubspotTraitsWithId.id)
+    }
     const tiersCountTraits = await countTiersForUserId(userId)
     segmentIo.identify({
```
Comment (jordanh) [inline src/server/utils/sendSegmentEvent.ts]: @mattkrick do we even need this call any more? Do you remember why we're calling identify() on the client side at all?

**Comment by jordanh** [comment]
@mattkrick this is ready for 👀

File: src/server/utils/sendSegmentEvent.ts (hunk 1 above)
Comment (mattkrick) [reply src/server/utils/sendSegmentEvent.ts]: i believe you said in order to put stuff into hubspot it had to be through an identify call

File: src/universal/components/AnalyticsIdentifier.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ class AnalyticsIdentifier extends Component<Props, State> {
       avatar,
       created,
       email,
-      name
+      parabolPreferredName: name
```
Comment (mattkrick) [inline src/universal/components/AnalyticsIdentifier.tsx]: i think your comment was meant for this one?
if so, i can't remember why we did this. maybe before all the ID calls on the server?
Seems safe enough to remove :man_shrugging:

Comment (jordanh) [reply src/universal/components/AnalyticsIdentifier.tsx]: Oh yes, correct. The comment was supposed to be for this one.

Here's my guess: this identify() call is the only place to send the anonymous segment id to later get alias()ed on the server. The only issue I noticed with this call is that it gets called for everybody quite often. It's generating extra events.

Any objection to me adding a // TODO: above this to look at it and consider limiting it to only sending when there is anonymous id to share with segment?

Comment (mattkrick) [reply src/universal/components/AnalyticsIdentifier.tsx]: nah, the segment id gets sent along with a signup, this never gets called for anonymous folks.
i'd be fine just removing the call to window.anlytics.identify & seeing what mayhem it causes.

Comment (jordanh) [reply src/universal/components/AnalyticsIdentifier.tsx]: Ok, I agree. Let's pull it and see what breaks.

**Pull request merged**